src/groupelements.jl

```diff
@@ -0,0 +1,143 @@
+# Sectors corresponding to irreducible representations of compact groups
+#------------------------------------------------------------------------------#
+"""
+    abstract type AbstractGroupElement{G<:Group,ω} <: Sector end
+
+Abstract supertype for sectors which corresponds to group elements of a group `G`, with
+the associator given by a 3-cocycle `ω`.
+
+Actual concrete implementations of those irreps can be obtained as `Element[G]`, or via their
+actual name, which generically takes the form `(asciiG)Element`, i.e. the ASCII spelling of
+the group name followed by `Irrep`.
+
+All group elements have [`FusionStyle`](@ref) equal to `UniqueFusion()`.
+For the fusion structure, a specific `SomeGroupElement<:AbstractElment{SomeGroup,ω}`
+should only implement the following methods
+```julia
+Base.:*(c1::GroupElement, c2::GroupElement) -> GroupElement
+Base.one(::Type{GroupElement}) -> GroupElement
+Base.inv(c::GroupElement) -> GroupElement
+TensorKitSectors.cocycle(c1::GroupElement, c2::GroupElement, c3::GroupElement) -> Number
+```
+and the methods `conj`, `⊗`, `Nsymbol`, `Fsymbol`, `dim`, `Asymbol`, `Bsymbol` and
+`frobeniusschur` will be automatically defined.
+"""
+abstract type AbstractGroupElement{G <: Group, ω} <: Sector end # irreps have integer quantum dimensions
+FusionStyle(::Type{<:AbstractGroupElement}) = UniqueFusion()
+
+⊗(a::I, b::I) where {I <: AbstractGroupElement} = (a * b,)
+Base.one(a::AbstractGroupElement) = one(typeof(a))
+Base.conj(a::AbstractGroupElement) = inv(a)
+Nsymbol(a::I, b::I, c::I) where {I <: AbstractGroupElement} = c == a * b
+function Fsymbol(a::I, b::I, c::I, d::I, e::I, f::I) where {I <: AbstractGroupElement}
+    ω = cocycle(a, b, c)
+    if e == a * b && d == e * c && f == b * c
+        return ω
+    else
+        return zero(ω)
+    end
+end
+dim(c::AbstractGroupElement) = 1
+frobeniusschur(c::AbstractGroupElement) = cocycle(c, inv(c), c)
+function Asymbol(a::I, b::I, c::I) where {I <: AbstractGroupElement}
+    A = frobeniusschur(dual(a)) * cocycle(inv(a), a, b)
+    return c == a * b ? A : zero(A)
+end
+function Bsymbol(a::I, b::I, c::I) where {I <: AbstractGroupElement}
+    B = cocycle(a, b, inv(b))
+    return c == a * b ? B : zero(B)
+end
+
+struct ElementTable end
+"""
+    const Element
```

Comment:
This is quite a generic name to use for this, I'm okay with keeping it if there isn't really something better, but I don't necessarily feel like `Element` obviously shows that it is a group-element.

How unreasonable is just using `GroupElement[G, ω]`? I did some quick browsing through wiki and nLab and also found the term `Grp`, but I'm not entirely sure if that is accurate.

Reply:
Sure, `GroupElement` is fine with me. `Vec[GroupElement[G]]` is going to be strange, but I wouldn't mind adding a `Vec[G] -> Vec[GroupElement[G]]` definition in TensorKit, unless you think this is too much heterodoxy?

Reply:
I fully agree with just writing `Vec[G,omega]` (I think you need the second argument), even if we decide on keeping `Element[G,omega]`, and I actually would say that this is the preferred way of constructing that, but 🤷